src/event.c

```diff
@@ -137,7 +142,10 @@ void process_input(struct device *dev, struct dev_input *inp)
 		}
 		sign = cfg.invert[inp->idx] ? -1 : 1;
 
-		inp->val = (int)((float)inp->val * cfg.sensitivity * (inp->idx < 3 ? cfg.sens_trans[inp->idx] : cfg.sens_rot[inp->idx - 3]));
+		float sens_rot = cfg.filter_rotation ? 0 : cfg.sens_rot[inp->idx - 3];
```

Comment:
Spacenavd is a C89 codebase. Variable delcarations are only valid at the top of the encolsing scope or function.

Reply:
ops, changing it